### PR TITLE
Check wg-public key format

### PR DESCRIPTION
**Pull request by chadek**
Adding a peer with a wrong public key format to a tunnel configuration makes the wall tunnel hang silently when trying to save configuration (and then restart the service). This should prevent wrong configuration but some checks may be done server side to, at least, log some error and display some warning.

### CODE REVIEW

File: net/wireguard/src/opnsense/mvc/app/models/OPNsense/Wireguard/Client.xml
Comment (fichtner) [inline net/wireguard/src/opnsense/mvc/app/models/OPNsense/Wireguard/Client.xml:18]: This isn't all proper base64. AFAIK the "=" can be zero to three times for padding. The 43 is probably a guess as well?

@AdSchellevis what do you think about a Base64Field type that validates this correctly?

Comment (AdSchellevis) [reply net/wireguard/src/opnsense/mvc/app/models/OPNsense/Wireguard/Client.xml:18]: @fichtner we can easily add a field type for it, which would be way better than spreading TextField's all over the place.

Comment (mimugmail) [reply net/wireguard/src/opnsense/mvc/app/models/OPNsense/Wireguard/Client.xml:18]: I like :)

Comment (AdSchellevis) [reply net/wireguard/src/opnsense/mvc/app/models/OPNsense/Wireguard/Client.xml:18]: it someone opens an issue in core, I'll promise to look into it.

Comment (fichtner) [reply net/wireguard/src/opnsense/mvc/app/models/OPNsense/Wireguard/Client.xml:18]: https://github.com/opnsense/core/issues/4398 thanks!

Comment (chadek) [reply net/wireguard/src/opnsense/mvc/app/models/OPNsense/Wireguard/Client.xml:18]: 43 + = isn't a guess: I basically copy paste the check perform for the psk field, ensuring the regex actually match well a key generated by wg pubkey. I like the idea of a Base64Field but I'm not familiar enough with opnsense src to have an idea of what is better. If Base64Field allow different size, it should be added something to check the string lengh

Comment (chadek) [reply net/wireguard/src/opnsense/mvc/app/models/OPNsense/Wireguard/Client.xml:18]: If this is to be done for this field, I suggest you also update the psk field format to Base64Field

Comment (AdSchellevis) [reply net/wireguard/src/opnsense/mvc/app/models/OPNsense/Wireguard/Client.xml:18]: @mimugmail https://github.com/opnsense/core/commit/43bd9812b681ae055201c28768052a436c6bb308 there you go.

File: net/wireguard/src/opnsense/mvc/app/models/OPNsense/Wireguard/Client.xml
Comment (fichtner) [inline net/wireguard/src/opnsense/mvc/app/models/OPNsense/Wireguard/Client.xml:18]: I don't think the mask is needed any longer as the type validates it, which leaves you with the length, but it seems like micromanagement to check it and break later when new key length is added

File: net/wireguard/src/opnsense/mvc/app/models/OPNsense/Wireguard/Client.xml
Comment (chadek) [reply net/wireguard/src/opnsense/mvc/app/models/OPNsense/Wireguard/Client.xml]: Sorry contributing is kind of new to me, don't know what are best practices here. I revert and also updated mask to match the 45 string

File: net/wireguard/src/opnsense/mvc/app/models/OPNsense/Wireguard/Client.xml
Comment (chadek) [reply net/wireguard/src/opnsense/mvc/app/models/OPNsense/Wireguard/Client.xml:18]: Agree but I think this check is currently useful until the wall wireguard server stop hang if one public key of one peer doesn't match the required format. I just tried to provide a random base64 string with the wrong size setting up one test peer and it prevented the wall wireguard process to restart, not just the associated tunnel "server". It is a real issue which must be fix somewhere else. Keeping this checks here for now may help in the mean time 

What do you think?

Comment (fichtner) [reply net/wireguard/src/opnsense/mvc/app/models/OPNsense/Wireguard/Client.xml:18]: the base64 field was added to reduce complexity and typographic errors in manual regex masks. having the field and keeping the mask amounts do doing the base64 field work for very little benefit and keeps maintenance of these configuration files higher than they have to be.

also think of copy+paste development so bad style like this will get carried over if we don't keep on top of it. :)

Comment (chadek) [reply net/wireguard/src/opnsense/mvc/app/models/OPNsense/Wireguard/Client.xml]: Yeah right, I also think this is enough check performed client-side, public key should be, somehow, checked server side and an issue may be opened. I mentioned in an other thread this PR #2008 which may help

**Review by fichtner — APPROVED** [review]
ok from me, thanks! waiting for final maintainer feedback now

**Review by mimugmail — APPROVED** [review]
Looks good, thanks. :)

**Comment by fichtner** [comment]
Merged, thanks!

**Pull request merged**
